### PR TITLE
[MM-59332] Migrate tooltips of "components/post_edit_history/edited_post_item/edited_post_item.tsx" to WithTooltip

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27560
Jira https://mattermost.atlassian.net/browse/MM-59332

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
@Aryakoste can you check the failing CI

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/post_edit_history/edited_post_item/__snapshots__/edited_post_item.test.tsx.snap [hunk 1]
```diff
@@ -85,24 +85,10 @@ exports[`components/post_edit_history/edited_post_item should match snapshot 1`]
...
+        <WithTooltip
+          id="editPostRestoreTooltip"
           placement="left"
-          trigger={
-            Array [
-              "hover",
-              "focus",
-            ]
-          }
+          title="Restore this version"
```
Comment (Rajat-Dabade) [inline webapp/channels/src/components/post_edit_history/edited_post_item/__snapshots__/edited_post_item.test.tsx.snap:91]: Should we use `Intl` `formatMessage` for the title?

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/post_edit_history/edited_post_item/__snapshots__/edited_post_item.test.tsx.snap:91]: This is snapshot it does used i18n

Comment (Rajat-Dabade) [reply webapp/channels/src/components/post_edit_history/edited_post_item/__snapshots__/edited_post_item.test.tsx.snap:91]: Aah, my bad.

**Review by Rajat-Dabade — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**